Bottom: a2a1f17dc6b908f0033d11b39e720dad30ac1393
Top:    5f2689fca0b3019d119d4c2a704cafec73cf205e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-24 01:05:56 +0000

action.c: Unconditionally notify NetlistChanged() when loading new netlist

We need to call NetlistChanged() even if the loading failed, otherwise
the GUI is not informed that we free'd the old netlist.


---

diff --git a/src/action.c b/src/action.c
index 665ae91..b1c33ab 100644
--- a/src/action.c
+++ b/src/action.c
@@ -5852,8 +5852,8 @@ ActionLoadFrom (int argc, char **argv, Coord x, Coord y)
 	free (PCB->Netlistname);
       PCB->Netlistname = StripWhiteSpaceAndDup (name);
       FreeLibraryMemory (&PCB->NetlistLib);
-      if (!ImportNetlist (PCB->Netlistname))
-	NetlistChanged (1);
+      ImportNetlist (PCB->Netlistname);
+      NetlistChanged (1);
     }
   else if (strcasecmp (function, "Revert") == 0 && PCB->Filename
 	   && (!PCB->Changed
